src/api/ruleMapping/ruleMapping.service.ts

```diff
@@ -42,16 +60,22 @@ export class RuleMappingService {
         }));
       }
     });
-    return { [fieldKey]: fields };
+
+    const results = await Promise.all(promises);
+    const fields = results.flat();
+
+    const uniqueFieldsMap = new Map<string, any>();
+
+    fields.forEach((field) => {
+      uniqueFieldsMap.set(field.property, field);
+    });
```

Comment:
I think you can just use map here: `const uniqueFieldsMap = new Map(fields.map(field => [field.property, field]));`

Reply:
Updated! Definitely cleaner, thanks :)